pysradb/search.py

```diff
@@ -0,0 +1,1619 @@
+            ("Library Selection",),
+            ("Platform",),
+            ("Organism", "Publication Date"),
+            ("Library Source", "Publication Date"),
+            ("Library Selection", "Publication Date"),
+            ("Platform", "Publication Date"),
+            ("Library Source", "Organism"),
+            ("Library Selection", "Organism"),
+            ("Platform", "Organism"),
+            ("Library Selection", "Library Source"),
+            ("Platform", "Library Source"),
+            ("Platform", "Library Selection"),
+        ]
+        plot_keys = {
+            "daterange": "Publication Date",
+            "organism": "Organism",
+            "source": "Library Source",
+            "selection": "Library Selection",
+            "platform": "Platform",
+            "basecount": "Base Count",
+        }
+        if "all" not in graph_types:
+            selected_plots = []
+            for graph_type in graph_types:
+                if graph_type not in plot_keys:
+                    continue
+                for plot in plots:
+                    if plot_keys[graph_type] in plot and plot not in selected_plots:
+                        selected_plots.append(plot)
+            plots = selected_plots
+        too_many_organisms = False
+        if self.stats["graph_raw"]["Organism"].nunique() > 30:
+            print(
+                "Too many types of organisms to plot (>30). Showing only top 30 organisms."
+            )
+            too_many_organisms = True
+        for plot in plots:
+            self._plot_graph(plt, plot, show, saveto, too_many_organisms)
+
+    def search(self):
+        pass
+
+    def get_df(self):
+        """Getter for the search result dataframe."""
+        return self.df
+
+    def _plot_graph(self, plt, axes, show, savedir, too_many_organisms):
+        """Plots a graph based on data from self.stats
+
+        Parameters
+        ----------
+        axes: tuple
+            tuple containing 1 or 2 strings corresponding to the statistics
+            to plot. 1 string: Histogram. 2 string: heat map
+        savedir: str
+            directory to save to
+        show: bool
+            whether to call plt.show
+        """
+
+        if (
+            "Publication Date" in axes
+            and self.stats["graph_raw"]["Publication Date"].nunique() > 30
+        ):
+            self.stats["graph_raw"]["Publication Date"] = self.stats["graph_raw"][
+                "Publication Date"
+            ].str[:-3]
+        if axes == ("Base Count",):
+            count = list(self.stats["count_data"])
+            title = "Histogram of Base Count"
+            plt.figure(figsize=(15, 10))
+            plt.hist(count, min(70, len(count)), color="#135c1c", log=True)
+            plt.xlabel("Base Count", fontsize=14)
+            plt.ylabel("Frequency", fontsize=14)
+            plt.xticks(rotation=90)
+            plt.title(title, fontsize=18)
+            plt.savefig(f"{savedir}{title}.jpeg")
```

Comment:
Besides saving, can you return the plotting object to the user, for example:
```python
fig, ax = plt.subplots()
ax.scatter(x,y)
return ax
```

Additionally we could also allow for the user to provide us an axes: `ax` that we can then use to create the same scatter plot. This gives the user a lot of flexibility to change titles, labels of axes and play around with colors

Reply:
resolved in 44ee73c